### PR TITLE
use tooltip for descriptions on entity pages

### PR DESCRIPTION
**Pull request by kdoh**
implements #2755

Per @mazameli's solution, If a dashboard or question has a description and you're on the main "detail" page for it, you'll see a little info button, if not then you'll just see the title as before.

screen shot 2016-09-06 at 1 28 26 pm

### CODE REVIEW

**Comment by kdoh** [comment]
cc @mazameli @tlrobinson and @salsakran for thoughts (and code review as appropriate) on this.

File: frontend/src/metabase/components/HeaderBar.jsx [hunk 1]
```diff
@@ -23,12 +24,12 @@ export default class Header extends Component {
                 </div>
             );
         } else {
-            if (name && description) {
+            if(name && description) {
```
Comment (mavimo) [inline frontend/src/metabase/components/HeaderBar.jsx]: Why do you remove space between if and ( ? I suppose coding standard need space..

File: frontend/src/metabase/components/HeaderBar.jsx [hunk 2]
```diff
@@ -23,12 +24,12 @@ export default class Header extends Component {
...
                 titleAndDescription = (
-                    <div className="Header-title my1 py2">
-                        <h2>{name}</h2>
-                        <h4 className="Header-title-name text-grey-3">{description || "No description yet"}</h4>
-                    </div>
+                    <TitleAndDescription
+                        title={ name }
```
Comment (mavimo) [inline frontend/src/metabase/components/HeaderBar.jsx]: I also suppose space should be removed between {} and prop name...

**Comment by kdoh** [comment]
@tlrobinson Did you have a chance to take a look at this?

**Comment by tlrobinson** [comment]
LGTM

**Comment by salsakran** [comment]
this behaves a bit weirdly with longish descriptions


screen shot 2016-09-07 at 1 43 52 pm

**Comment by kdoh** [comment]
If we're generally ok with the tooltip approach one  solution could be to increase the width of the tooltips in this case to make the line length reasonable which should help. @mazameli what do you think since this was your baby?

**Comment by mazameli** [comment]
Yeah, maybe something like `max-width: 400px`, and bump up the `line-height` to 1.5x-ish the font size. For evaluation, you could put in an amount of text roughly equal to that example Sameer put in Slack yesterday (the description he wrote on that new card).

**Pull request merged**